adapters/repos/db/shard_write_merge.go

```diff
@@ -53,9 +53,14 @@ func (s *Shard) MergeObject(ctx context.Context, merge objects.MergeDocument) er
 		}
 	}
 
-	if len(merge.Vector) > 0 && s.hasLegacyVectorIndex() {
+	if len(merge.Vector) > 0 {
+		vectorIndex, ok := s.GetVectorIndex("")
+		if !ok {
+			return errors.Errorf("Validate vector index for update of %v for vector: vector index not found", merge.ID)
```

Comment:
```suggestion
			return errors.Errorf("validate vector index for update of %v for vector: vector index not found", merge.ID)
```

See the relevant [Code Review Comment](https://go.dev/wiki/CodeReviewComments#error-strings)

Reply:
Yeah, agree. I was following other errors in this file, but it is better to clean them up as well than to follow